theia_tests.Jenkinsfile

```diff
@@ -8,123 +8,96 @@
       }
-    },
-    timeout: [
-        time: 20,
-        unit: 'MINUTES'
-    ]
+    }
   )
 
   pipeline.createStage(
-    name: "Build vsix as plugin",
-    stage: {
-      //Run build
-      sh "cp resources/testProfileData.example.ts resources/testProfileData.ts"
-      pipeline.nvmShell "npm run build"
-      
-      // Gather details for build archives
-      def vscodePackageJson = readJSON file: "package.json"
-      def date = new Date()
-      String buildDate = date.format("yyyyMMddHHmmss")
-      def fileName = "vscode-extension-for-zowe-v${vscodePackageJson.version}-${BRANCH_NAME}-${buildDate}"
-
-      // Generate a vsix for archiving purposes
-      pipeline.nvmShell "npx vsce package -o ${fileName}.vsix"
-
-      dir ("/tmp/theia/theia") {
-        sh "mkdir -p plugins"
-      }
-      // Copy vsix to Theia plugins folder
-      sh "cp ${fileName}.vsix '/tmp/theia/theia/plugins/${fileName}.vsix'"
+    name          : "Start Theia",
+    timeout       : [ time: 10, unit: 'MINUTES' ],
+    isSkippable   : true,
+    stage         : {
+      echo ">> Start Thiea in docker"
+      sh "docker run -d -p 3000:3000 -v \"\$(pwd)/${packagingDir}:/home/theia/.theia\" -u theia theiaide/theia"
+
+      echo ">> Wait for 2 minutes to give time for Theia to be started"
+      sleep time: 2, unit: 'MINUTES'
+
+      echo ">> Try to access localhost:3000 and see if it's available"
+      sh "firefox -headless --screenshot ${packagingDir}/screenshots/firefox-test.jpg http://localhost:3000/"
 
-      // Upload vsix to Artifactory
-      withCredentials([usernamePassword(credentialsId: ARTIFACTORY_CREDENTIALS_ID, usernameVariable: 'USERNAME', passwordVariable: 'PASSWORD')]) {
-        def uploadUrlArtifactory = "${ARTIFACTORY_UPLOAD_URL}/${fileName}.vsix"
-        sh "curl -u ${USERNAME}:${PASSWORD} --data-binary \"@${fileName}.vsix\" -H \"Content-Type: application/octet-stream\" -X PUT ${uploadUrlArtifactory}"
+      echo ">> Show container logs"
+      sh "docker logs \$(docker ps -q)"
+
+      echo ">> Show files in theia working folder"
+      sh "find ${packagingDir} -type f -maxdepth 3"
+    }
+  )
+
+  pipeline.test(
+    name          : 'Theia',
+    timeout       : [ time: 10, unit: 'MINUTES' ],
+    operation     : {
+      ansiColor('xterm') {
+        pipeline.nvmShell "npm run test:theia"
       }
     },
-    timeout: [
-        time: 10,
-        unit: 'MINUTES'
-    ]
+    // FIXME: once we publish test result as junit, we can disable below line
+    allowMissingJunit : true
+    // example way to define junit file
+    // junit         : "path/to/junit.xml",
+    // example way to define html report
+    // htmlReports   : [
+    //   [dir: "path/to/html/reports", files: "index.html", name: "Report: Theia Test"],
+    // ],
   )
 
-  pipeline.createStage(
-    name: "Check Firefox",
-    stage: {
-        sh "firefox --version | more"
+  // define we need publish stage
+  pipeline.publish(
```

Comment:
It'd be nice it the publish stage wasn't a requirement (since it's not needed in this situation

Reply:
The publish stage is not mandatory. So if we don't need to publish, we can just comment or remove this stage.